src/js/components/DataTable/Body.js

```diff
@@ -179,22 +180,41 @@ const Body = forwardRef(
     const [active, setActive] = React.useState();
     const [lastActive, setLastActive] = React.useState();
 
+    const selectRow = () => {
+      const primaryValue = data[active]?.[primaryProperty];
+      if (selected && selected.includes(primaryValue)) {
+        onSelect(selected.filter((s) => s !== primaryValue));
+      } else onSelect([...selected, primaryValue]);
+    };
+
+    const clickedRow =
+      onClickRow &&
+      active >= 0 &&
+      (!disabled ||
+        !disabled.includes(datumValue(data[active], primaryProperty)));
+
     return (
       <Keyboard
-        onEnter={
-          // active is undefined if user never used the keyboard
-          onClickRow &&
-          active >= 0 &&
-          (!disabled ||
-            !disabled.includes(datumValue(data[active], primaryProperty)))
-            ? (event) => {
-                event.persist();
-                const adjustedEvent = event;
-                adjustedEvent.datum = data[active];
-                onClickRow(adjustedEvent);
-              }
-            : undefined
-        }
+        onEnter={(event) => {
```

Comment:
Can we only bind an event handler if the condition is true, rather than always binding a function and then testing whether it should do anything? For example: `onEnter={clickedRow ? (event) => { if ... } : undefined`

Reply:
done